pkg/agent/agent.go

```diff
@@ -279,57 +276,31 @@ func (k *klocksmith) process(ctx context.Context) error {
 		klog.Info("Node already marked as unschedulable")
 	}
 
-	klog.Info("Getting pod list for deletion")
-
-	pods, err := k.getPodsForDeletion(ctx)
-	if err != nil {
-		return fmt.Errorf("getting list of pods for deletion: %w", err)
-	}
-
-	// Delete the pods.
-	// TODO(mischief): Explicitly don't terminate self? we'll probably just be a
-	// Mirror pod or daemonset anyway..
-	klog.Infof("Deleting %d pods", len(pods))
-
-	for _, pod := range pods {
-		klog.Infof("Terminating pod %q...", pod.Name)
-
-		if err := k.pg.Pods(pod.Namespace).Delete(ctx, pod.Name, metav1.DeleteOptions{}); err != nil {
-			// Continue anyways, the reboot should terminate it.
-			klog.Errorf("Failed terminating pod %q: %v", pod.Name, err)
-		}
+	dh := drain.Helper{
+		Ctx:                 ctx,
+		Client:              k.clientset,
+		Force:               false,
+		GracePeriodSeconds:  -1,
+		IgnoreAllDaemonSets: true,
+		DeleteEmptyDirData:  true,
+		Out:                 klogOutWriter{},
+		ErrOut:              klogErrWriter{},
 	}
 
-	// Wait for the pods to delete completely.
-	//
-	//nolint:varnamelen // Conventional name.
-	wg := sync.WaitGroup{}
-
-	for _, pod := range pods {
-		wg.Add(1)
-
-		go func(pod corev1.Pod) {
-			klog.Infof("Waiting for pod %q to terminate", pod.Name)
-
-			if err := k.waitForPodDeletion(ctx, pod); err != nil {
-				klog.Errorf("Skipping wait on pod %q: %v", pod.Name, err)
-			}
-
-			wg.Done()
-		}(pod)
+	klog.Info("Getting pod list for deletion")
+	pods, errs := dh.GetPodsForDeletion(k.nodeName)
+	if err != nil {
+		return fmt.Errorf("error getting pods for deletion: %v", errs)
 	}
 
-	allPodsTerminated := make(chan struct{})
-
-	go func() {
-		wg.Wait()
-		allPodsTerminated <- struct{}{}
-	}()
+	toDelete := k8sutil.FilterPods(pods.Pods(), func(p *corev1.Pod) bool {
+		return p.Namespace != "kube-system"
+	})
 
-	select {
-	case <-allPodsTerminated:
-	case <-ctx.Done():
-		return fmt.Errorf("waiting for all pods to terminate before the reboot interrupted")
+	klog.Infof("Deleting/Evicting %d pods", len(toDelete))
+	err = dh.DeleteOrEvictPods(toDelete)
+	if err != nil {
```

Comment:
Shorter notation should be preferred.
```suggestion
	if err := dh.DeleteOrEvictPods(toDelete); err != nil {
```

Reply:
done